arduino/build_arduino_package.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+#Syntax: ./build_arduino_package [-d] SKETCHBOOK_PATH
+#Adding -d builds a developer package by creating symlinks to the original files instead of copies
+#SKETCHBOOK_PATH is the path to the sketchbook for your Arduino IDE installation
+#On Linux this is usually ~/Arduino/
+#On Windows this is usually ~/Documents/Arduino/
+#To run on windows, use a bash emulator. This script was tested using git bash
+#NOTE: Script will not execute properly if any file paths have any spaces in them
+#NOTE: The developer build option does not work on Windows
+#Set paths to source files
+CONFIG_PATH="config"
+PACKAGE_PATH="packages/1.6.15"
+LIBSIGNPOST_PATH="../apps/libsignpost"
+MBEDTLS_PATH="../apps/support/mbedtls/mbedtls"
+
+#Process command line arguments
+case "$#" in
+	"1")
+		if [ ! -d "$1" ] ; then
+			echo "${1} is a not a directory, exiting..."
+			exit 1
+		fi
+		SKETCHBOOK_PATH="$1"
+		DEVELOPER=
+		echo "Building consumer package"
+		;;
+	"2")
+		#Check to see which argument is the sketchbook path
+		#If neither of the arguments are directories, exit
+		#If neither of the arguments are "-d", exit
+		if [ -d "$1" ] && [ "$2" = "-d" ] ; then
+			SKETCHBOOK_PATH="$1"
+		elif [ -d "$2" ] && [ "$1" = "-d" ] ; then
+			SKETCHBOOK_PATH="$2"
+		else
+			if [ ! -d "$1" ] && [ ! -d "$2" ] ; then
+				echo "Neither input argument is a directory, exiting..."
+			else
+				echo "Invalid syntax, exiting..."
+			fi
+			exit 1
+		fi
+		DEVELOPER="-sR"
```

Comment:
Consider a more clear variable name here given that this is changing `cp` flags. I expected this to be an optional package author name or some such